### ISSUE TITLE
Add Calendar HeatMap

### ISSUE THREAD

**Issue by chimeno** [ankane/chartkick.js#24]
It would be great to add a calendar chart to this awesome library.
Docs:
https://developers.google.com/chart/interactive/docs/gallery/calendar?hl=en
http://www.highcharts.com/demo/heatmap
I'll be trying to implement this

**Comment by alfredmey** [comment]
Any Progress on this... I Need this As Well

**Comment by buren** [comment]
I've started working on this a bit and I've got Highcharts working (see diff: https://github.com/ankane/chartkick.js/commit/bf66b020948aadf54243197138d1c4f488ca620c, though that is based of my fork and _very_ unfinished..)

I'm not sure its a good idea to expose google "calendar"-chart and highcharts heatmap chart under the same public API, since they're quite different.. One idea would be to expose a highcharts only `Heatmap` chart and a google only `Calendar`-chart. (I'm also having some trouble getting the callback to fire when loading the "calendar" package from google api:s.) It seems like it would be possible to build the equivalent of a google-calendar chart using highcharts heatmap, so we could probably also implement the "calendar"-chart  for the highcharts adapter. 

Thought / ideas anyone?

**Comment by buren** [comment]
https://github.com/ankane/chartkick.js/commit/1b606a5 implements both of these (with some caveats for the time being).

* Google only has a "calendar year layout"
* Highcharts has a heatmap implementation (matrix layout with the value as a "weight")
  - Got some date grouping functionality working

See screenshots:

<img width="709" alt="screen shot 2015-09-02 at 21 35 29" src="https://cloud.githubusercontent.com/assets/922411/9642095/9c59cfba-51ba-11e5-9d0b-7efedc56eaad.png">
<img width="702" alt="screen shot 2015-09-02 at 21 35 20" src="https://cloud.githubusercontent.com/assets/922411/9642096/9c5b3080-51ba-11e5-9e6d-134e9e9357a2.png">
<img width="707" alt="screen shot 2015-09-02 at 21 35 10" src="https://cloud.githubusercontent.com/assets/922411/9642097/9c5c73d2-51ba-11e5-8888-b5d8315c5c7b.png">

**Comment by buren** [comment]
See PR #49

**Comment by buren** [comment]
See PR #50 for Google Calendar visualization

**Comment by ankane** [comment]
Hey all, just fyi: there are no plans to support additional chart types at the moment.

**Comment by xreality** [comment]
I also looking for a calender chart and found nothing usable one. So I tried something for myself and this came out: https://jsfiddle.net/xreality/410t5a8j/
Both axis types are categories, so the months are placed as blocks in the matrix. This can also be implemented programmatically with offsets quite easily. Then the start and end dates of the calendars can be called up and it is decided whether all months should be displayed in a row or with a break (as here).
Unfortunately, the month names from the series are not drawn, so I had to add them as a Plotline label.
